### PR TITLE
Snake act

### PR DESCRIPTION
**Pull request by nithinraok**
# What does this PR do ?

Add snake activation to Audio Codec

**Collection**: TTS/ASR

# Changelog 
- Add Snake Activation to activation file
- Optionally choose between ELU/Snake while building model
- Note: Prior checkpoints don't load as is without modifying variable names 

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/collections/tts/modules/encodec_modules.py [hunk 1]
```diff
@@ -59,13 +60,33 @@
 from nemo.utils.decorators import experimental
 
 
+class CodecActivation(nn.Module):
```
Comment (rlangman) [inline nemo/collections/tts/modules/encodec_modules.py:63]: Can we move this to `audio_codec_modules.py`? I think we should avoid adding code to `encodec_modules.py` that is not part of the original encodec source.

File: nemo/collections/asr/parts/utils/activations.py [hunk 2]
```diff
@@ -12,11 +12,34 @@
...
 # limitations under the License.
 
+import torch
 import torch.nn as nn
 
 __all__ = ['Swish']
 
 
+# @torch.jit.script
+def snake(x, alpha: torch.Tensor):
```
[the remark this answers is not in the document]
Comment (nithinraok) [reply nemo/collections/asr/parts/utils/activations.py:22]: Done.

File: nemo/collections/asr/parts/utils/activations.py [hunk 3]
```diff
@@ -12,11 +12,34 @@
...
+    x = x.reshape(shape)
+    return x
+
+
+class Snake(nn.Module):
+    """
+    Snake activation function introduced in 'https://arxiv.org/abs/2006.08195'
+    """
+
+    def __init__(self, channels):
```
[the remark this answers is not in the document]
Comment (nithinraok) [reply nemo/collections/asr/parts/utils/activations.py:35]: Done

File: examples/tts/conf/audio_codec/encodec_24000.yaml [hunk 4]
```diff
@@ -87,10 +87,10 @@ model:
   # Remove this section to disable logging.
   log_config:
     log_dir: ${log_dir}
-    log_epochs: [10, 50]
+    log_epochs: [10, 50, 100, 150, 200]
     epoch_frequency: 100
     log_tensorboard: false
-    log_wandb: false
+    log_wandb: true
```
[the remark this answers is not in the document]
Comment (nithinraok) [reply examples/tts/conf/audio_codec/encodec_24000.yaml:93]: yep, I will also change the frequency. However this should be later modified to use steps than epochs.

File: nemo/collections/asr/parts/utils/activations.py [hunk 5]
```diff
@@ -12,11 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import torch
 import torch.nn as nn
 
 __all__ = ['Swish']
 
 
+# @torch.jit.script
```
[the remark this answers is not in the document]
Comment (nithinraok) [reply nemo/collections/asr/parts/utils/activations.py:21]: I fixed it for jit script, now code runs a bit faster. when tested on 100 steps per epoch it showed 8% improvement over non-jit scripted activation.

File: nemo/collections/tts/modules/encodec_modules.py (hunk 1 above)
Comment (nithinraok) [reply nemo/collections/tts/modules/encodec_modules.py:63]: Sure, good point.

**Review by rlangman — APPROVED** [review]

**Pull request merged**